aiohue/bridge.py

```diff
@@ -85,13 +104,73 @@ async def request(self, method, path, json=None, auth=True):
                 _raise_on_error(data)
                 return data
 
-        except ClientConnectionError:
+        except client_exceptions.ClientConnectionError:
             if self.proto is not None:
                 raise
 
             self.proto = "http"
             return await self.request(method, path, json, auth)
 
+    async def request_2(self, method, path):
+        """Make a request to any path with Hue's new request method.
+
+        This method has the auth in a header.
+        """
+        url = f"{self.proto}://{self.host}/{path}"
+
+        async with self.websession.request(
+            method,
+            url,
+            ssl=False,
+            headers={"hue-application-key": self.username},
+        ) as res:
+            res.raise_for_status()
+            return await res.json()
+
+    async def listen_events(self):
+        """Listen to events and apply changes to objects."""
+        loop = asyncio.get_running_loop()
+        pending_events = asyncio.Queue()
+
+        async def receive_events():
+            while True:
+                self.logger.debug("Subscribing to events")
+                try:
+                    for event in await self.clip.next_events():
+                        self.logger.debug("Received event: %s", event)
+                        pending_events.put_nowait(event)
+                except client_exceptions.ClientResponseError as err:
+                    self.logger.debug("Event endpoint %s", err.status)
+                    if err.status == 503:
+                        self.logger.debug("Sleeping while waiting for 503 to resolve")
+                        await asyncio.sleep(1)
+                except asyncio.TimeoutError:
+                    pass
+
+        event_task = loop.create_task(receive_events())
+        try:
+            while True:
+                event = await pending_events.get()
+
+                if event["type"] != "update":
+                    self.logger.debug("Unknown event type: %s", event)
+                    continue
+
+                for update in event["data"]:
+                    item_type = update["id_v1"].split("/", 2)[1]
+
+                    if item_type == "lights":
+                        obj = self.lights.process_update_event(update)
+                        # if obj is None, we didn't know the object
+                        # We could consider triggering a full refresh
+                        if obj is not None:
+                            yield obj
+
+        except asyncio.CancelledError:
```

Comment:
Would it be safe to only wrap the `await pending_events.get()` with the try... except block? We would decrease the indentation a bit for readers.